scribejava-apis/src/main/java/com/github/scribejava/apis/FlickrApi.java

```diff
@@ -10,7 +10,16 @@
  */
 public class FlickrApi extends DefaultApi10a {
 
-    protected FlickrApi() {
+    private static final String AUTHORIZE_URL = "https://www.flickr.com/services/oauth/authorize?oauth_token=%s";
+
+    private String perm; /* read, write, or delete (delete includes read/write) */
```

Comment:
See: https://www.flickr.com/services/api/auth.oauth.html

```
https://www.flickr.com/services/oauth/authorize

After getting the Request Token, your application needs to present the Flickr authorization page to the user, where they will be asked to give permission to your application to access their data. The authorization page will present the user with a list of permissions your application is requesting, as defined in the authentication flow settings.

Additionally, you can pass the optional perms= parameter, asking for read, write, or delete privileges. This parameter will override the setting defined in your application's authentication flow.
```

I think given the lack of an interface definition, defining read()/write()/delete() is confusing.  I created an enum as the constructor now.